.github/pull_request_template.md

```diff
@@ -0,0 +1,50 @@
+# Description
+Mandatory, single line description of the pull request, written as though it was an order.
+
+Then, following a single blank line, a summary of the change and which issue is fixed, including 
+- [ ] what is the problem being solved
+- [ ] why this is a good approach
+- [ ] what are the shortcomings of this approach, if any
+It may include
+- [ ] background information, such as "Fixes #"
+- [ ] benchmark results
+- [ ] links to design documents
+- [ ] dependencies required for this change
+
+The summary may omit some of these details if this PR fixes a single ticket that includes these details. It is the reviewer's discretion. 
+
+## [Optional] Changes
+Multiple changes are not recommended, so this section should normally be omitted. Unforunately, they are sometimes unavoidable. If there are multiple logical changes, list them separately.
+
+1. `'foo'` is replaced by `'bar'`
+2. `'fizzbuzz'` is optimized for gas
+
+## Type of change
+
+Feel free to delete options that are not relevant.
+
+- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
+- [ ] Bug fix (non-breaking change which fixes an issue)
+- [ ] New feature (non-breaking change which adds functionality)
+- [ ] Chore (tech debt, improved developer experience, etc)
+- [ ] Performance improvement (either gas efficiency or computational efficiency)
+- [ ] Requires a documentation update
+
+# [Optional] How Has This Been Tested?
+
+Did you need to run manual tests to verify this change? If so, please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
+
+# Checklist:
+
+## Code quality
+- [ ] I have written clear commit messages
```

Comment:
Does this apply if we're planning on squashing and merging? 😛  

Or should we avoid doing that?

Reply:
Clear commit messages are still useful, for the reviewer, posterity, and the author of the squashed commit:
1. the reviewer will want to be able to view the commits separately. If a file is moved, and then a functional change is introduced, it is important for the reviewer to view a diff that does not mix the file rename + functional change
2. posterity can still do the same, with squashed commits, by navigating to this PR. If this functionality were present, I would be strongly _against_ squash & merge
3. the author of the squashed commit wants to use the commit messages as a skeleton of the squashed commit's message